src/MongoDB/Manager.c

```diff
@@ -669,11 +670,46 @@ static PHP_METHOD(Manager, startSession)
 		return;
 	}
 
-	if (options && php_array_exists(options, "causalConsistency")) {
+	if (options && php_array_existsc(options, "causalConsistency")) {
 		cs_opts = mongoc_session_opts_new();
 		mongoc_session_opts_set_causal_consistency(cs_opts, php_array_fetchc_bool(options, "causalConsistency"));
 	}
 
+	if (options && php_array_existsc(options, "defaultTransactionOptions")) {
+		zval* txn_options = php_array_fetch_array(options, "defaultTransactionOptions");
+		mongoc_transaction_opt_t* txn_opts = NULL;
+
+		/* Thrown exception and return if the defaultTransactionOptions is not an array */
+		if (!txn_options) {
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected \"defaultTransactionOptions\" option to be an array");
+			if (cs_opts) {
+				mongoc_session_opts_destroy(cs_opts);
+			}
+			return;
```

Comment:
If we restructure the end of this function, can we utilize a `cleanup:` label? Consider:

```c
	cs = mongoc_client_start_session(intern->client, cs_opts, &error);

	if (cs) {
		phongo_session_init(return_value, cs TSRMLS_CC);
	} else {
		phongo_throw_exception_from_bson_error_t(&error TSRMLS_CC);
	}

cleanup:
	if (cs_opts) {
		mongoc_session_opts_destroy(cs_opts);
	}
}
```

Then we can use `goto cleanup;` here and within the `if (EG(exception))` block below.

Reply:
Yes, and a little more too.